### PR TITLE
Fixing manifests

### PR DESCRIPTION
**Pull request by piontec**
This PR does fixes for all the CAPI based clusters (vintage AWS ignored as we won't support them). It also adds assertion manifests, which check if the real status on the cluster is matching the assertion manifests, so that we can be sure that we get what we expect for our "working perfectly" examples.

The PR is huge, but the majority of changes are about added assertion manifests.

### CODE REVIEW

File: bases/environments/stages/prod/hello_app_cluster/kustomization.yaml [hunk 1]
```diff
@@ -4,12 +4,14 @@ configMapGenerator:
...
       - values=hello_world_app_user_config.yaml
-    name: ${cluster_id}-hello-world-user-config
+    name: ${cluster_name}-hello-world-user-config
+    namespace: org-${organization}
   - behavior: replace
     files:
       - values=cluster_user_config.yaml
     name: ${cluster_name}-user-config
     namespace: org-${organization}
+    namespace: org-${organization}
```
Comment (mproffitt) [inline bases/environments/stages/prod/hello_app_cluster/kustomization.yaml:14]: Duplicate namespace

Comment (mproffitt) [inline bases/environments/stages/prod/hello_app_cluster/kustomization.yaml]: another duplicate namespace

**Comment by mproffitt** [comment]
In some of the examples, the namespace is set to `org-org-name` whilst in others, namespace is set to `org-org-name` and organisation is set to `org-name` whereby I believe in the MCs, both namespace and organisation are set to `org-name` and I think the extra `org-` is confusing.

A good example of this is https://github.com/giantswarm/gitops-template/blob/65160f0c69ede3377c800f23794ce0397dce1a8d/tests/ats/assertions/exists/organizations/workload-clusters/hello-app-staging-1/configmaps.yaml where you see both styles in use.

**Review by mproffitt — APPROVED** [review]

**Comment by piontec** [comment]
> In some of the examples, the namespace is set to `org-org-name` whilst in others, namespace is set to `org-org-name` and organisation is set to `org-name` whereby I believe in the MCs, both namespace and organisation are set to `org-name` and I think the extra `org-` is confusing.
> 
> A good example of this is https://github.com/giantswarm/gitops-template/blob/65160f0c69ede3377c800f23794ce0397dce1a8d/tests/ats/assertions/exists/organizations/workload-clusters/hello-app-staging-1/configmaps.yaml where you see both styles in use.

I think this is OK, as for an organization named "X" we create a namespace "org-X".

**Pull request merged**